app/javascript/runtimeBanner/RuntimeBanner.jsx

```diff
@@ -0,0 +1,146 @@
+/* global Runtime */
+
+import { h, Component } from 'preact';
+
+const BANNER_DISMISS_KEY = 'runtimeBannerDismissed';
+
+/**
+ * A banner that will be displayed to provide a deep link into a specified
+ * ConsumerApp based on the Runtime context. If the banner is dismissed it will
+ * keep track of this in localStorage so it won't be rendered again.
+ */
+export class RuntimeBanner extends Component {
+  dismissBanner = () => {
+    localStorage.setItem(BANNER_DISMISS_KEY, true);
+    this.removeFromDOM();
+  };
+
+  removeFromDOM = () => {
+    const container = document.getElementById('runtime-banner-container');
+    container?.remove();
```

Comment:
I never noticed that we use optional chaining in our JS codebase 😍

Reply:
I absolutely love it too! I've had the linter (or something else) complain about using it in the asset pipeline initializers, maybe bc the JS doesn't go through webpack? Seems safe to use here though 🚀